### PR TITLE
refactored camperbot rooms to single array and removed bad rooms

### LINKED ISSUE #185: Bot is down in /HelpBonfires - 10/26
Bot is down in /HelpBonfires and somewhat flakey in some other channels.  Needs to be kicked.

### PR DESCRIPTION
**Pull request by joepurdy**
closes #175 

I condensed the camperbot rooms to a single array at @abhisekp's suggestion and removed the city rooms. I also removed additional rooms that were throwing 404s. I removed the prefixChannelName function completely since everything is hardcoded in a single array now.

Also I added FreeCodeCamp/HelpBonfires to close #185 

I found that the joinBonfireRooms function call in nap/lib/bot/GBot.js was causing issues with neverending unable to join errors due to the bonfire rooms already being added in the map() function on ln 267 of nap/data/RoomData.js, so I commented it out. Not sure if this is what you're looking for, but it's what I've got right now.

Let me know if this isn't right and I can tweak it more.

cc @dcsan

### CODE REVIEW

File: nap/data/RoomData.js [hunk 1]
```diff
@@ -128,476 +120,48 @@ var bonfireDashedNames = [
...
+    "camperbot/HelpZiplines",
+    "camperbot/devteam",
+    "camperbot/testing",
+    "FreeCodeCamp/40PlusDevs",
+    "FreeCodeCamp/Beta",
+    "FreeCodeCamp/BookClub",
+    "FreeCodeCamp/Casual",
+    "FreeCodeCamp/CodeReview",
+    "FreeCodeCamp/CodingJobs",
+    "FreeCodeCamp/CurriculumDevelopment",
```
Comment (abhisekp) [inline nap/data/RoomData.js]: No bot required in **CurriculumDevelopment** room

**Comment by abhisekp** [comment]
@dcsan why do you think it's not needed?

**Comment by abhisekp** [comment]
@dcsan this is indeed needed along with the regex fixes!!

**Comment by joepurdy** [comment]
Are you sure we don't want it in CurriculumDevelopment @abhisekp ?

Technically if we go off the [Official List](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/Official-Free-Code-Camp-Chat-Rooms) CurriculumDevelopment is included. I get that that room is more for admin discussion, but it'd probably be nice to focus on supporting the official list above all else.

On that note though I wasn't sure about the off topic rooms from ln 151 to ln 157 as well as some other random rooms. Theses are all the rooms in the array not included on the official list anymore:

FreeCodeCamp/40PlusDevs
FreeCodeCamp/Beta
FreeCodeCamp/BookClub
FreeCodeCamp/Design
FreeCodeCamp/Issues
FreeCodeCamp/LetsPair
FreeCodeCamp/News
FreeCodeCamp/NonprofitProjects
FreeCodeCamp/TeamViewer
FreeCodeCamp/Wiki
FreeCodeCamp/GameDev
FreeCodeCamp/Gaming
FreeCodeCamp/Hardware
FreeCodeCamp/Music
FreeCodeCamp/Saving
FreeCodeCamp/SelfImprovement
FreeCodeCamp/TVandMovies

Should the bot be in any of these rooms?

Also I borked my fork (it's why it says the changes are in an unknown repository) after submitting this so I'll have to close this PR and open a new one to make more edits. Which is fine by me. I just want to be sure that it's worth drafting an updated PR and that we're in consensus on what should or shouldn't be changed.

cc: @dcsan

**Comment by abhisekp** [comment]
@joepurdy, @QuincyLarson has to decide in which rooms the bot is not required.

**Comment by joepurdy** [comment]
This PR is redundant now that I've opened https://github.com/FreeCodeCamp/camperbot/pull/6 on the new repo

**Pull request closed without merging**